src/persist/src/storage.rs

```diff
@@ -16,8 +16,46 @@ use std::str::FromStr;
 
 use abomonation_derive::Abomonation;
 use async_trait::async_trait;
+use futures_executor::block_on;
 
 use crate::error::Error;
+use crate::indexed::encoding::BlobMeta;
+
+/// Sanity check whether we can decode the Blob's persisted meta object.
+///
+/// TODO: this is a hack and we will need to get rid of this once we have a
+/// proper backwards compatibility policy.
+pub fn check_meta_version<B: Blob>(b: &mut B) -> Result<(), Error> {
+    let meta = match block_on(b.get("META"))? {
+        None => return Ok(()),
+        Some(bytes) => bytes,
+    };
+
+    let current_version = BlobMeta::version();
+    let persisted_version = BlobMeta::encoded_version(&meta)
+        .ok_or_else(|| Error::from("unable to determine previously persisted encoding version."))?;
+
+    if current_version == persisted_version {
+        // Nothing to do here, everything is working as expected.
+        Ok(())
+    } else if current_version > persisted_version {
+        // Delete all the keys, as we are upgrading to a new version.
+        let keys = block_on(b.list_keys())?;
+        for key in keys {
+            block_on(b.delete(&key))?;
+        }
+
+        Ok(())
+    } else {
+        // We are reading a version further in advance than current,
+        // likely because a user has downgraded to an older version of
+        // Materialize.
+        Err(Error::from(format!(
+            "Invalid persistence version found {} can only read {}. Exiting.",
```

Comment:
this should probably suggest the workaround: manually deleting the persist data

Reply:
nit: our other errors seem to start with lower-case

Reply:
done to both, though adding the workaround makes the test string a bit long -- i think thats fine?

Reply:
yeah fine by me